### PR TITLE
gh-125588: Teach the python PEG generator the new f-string tokens

### PR DESCRIPTION
**Pull request by pablogsal**
* Issue: gh-125588

### CODE REVIEW

**Comment by Eclips4** [comment]
The CI is failing because it is running mypy on Python 3.11, where `token.py` doesn't have `FSTRING_` variables. I think we should bump the Python version for this `mypy` workflow.

**Comment by Eclips4** [comment]
I've opened a #125592 for that mypy problem.

File: Tools/peg_generator/mypy.ini [hunk 1]
```diff
@@ -4,7 +4,7 @@ pretty = True
 show_traceback = True
 
 # Make sure the peg_generator can be run using Python 3.10:
```
Comment (AlexWaygood) [inline Tools/peg_generator/mypy.ini]: This comment needs to be updated.

(Is it no longer a goal to be able to run the peg_generator code using Python 3.10? Is it okay to require Python 3.12+ in order to build the CPython `main` branch? Should this be discussed more widely first?)

**Comment by Eclips4** [comment]
Merging main to get changes from #125592

File: Tools/peg_generator/mypy.ini (hunk 1 above)
Comment (pablogsal) [reply Tools/peg_generator/mypy.ini]: Hummmm, I can change the PR to (optionally expect those tokens so we remain compatible with 3.10+).

Comment (pablogsal) [reply Tools/peg_generator/mypy.ini]: Done

**Pull request merged**